File: refactored/assertion-roulette/java-junit/Should_allow_the_removal_and_addition_of_new_value_retrievers.java
import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.Test;

public class IExistsForTestingValueRetrieving {

    @Test
    public void shouldAllowTheRemovalAndAdditionOfNewValueRetrievers() {
        {
            Service service = new Service();
            for (ValueRetriever valueRetriever : service.getValueRetrievers().toArray(new ValueRetriever[0])) {
                service.getValueRetrievers().unregister(valueRetriever);
                assertFalse("Explanation message", service.getValueRetrievers().contains(valueRetriever));
            }

            IExistsForTestingValueRetrieving thing = new IExistsForTestingValueRetrieving();
            service.getValueRetrievers().register(thing);
            assertEquals("Explanation message", 1, service.getValueRetrievers().size());
            assertSame("Explanation message", thing, service.getValueRetrievers().get(0));
        }
    }
}

File: refactored/assertion-roulette/CSharpXUnit/CSharpXUnit.Tests/Tests/Should_allow_the_removal_and_addition_of_new_value_retrievers.cs
using Xunit;

public class IExistsForTestingValueRetrieving 
{
    [Fact]
    public void shouldAllowTheRemovalAndAdditionOfNewValueRetrievers() 
    {
        {
            var service = new Service();
            foreach (var valueRetriever in service.GetValueRetrievers().ToArray())
            {
                service.GetValueRetrievers().Unregister(valueRetriever);
                Assert.False(service.GetValueRetrievers().Contains(valueRetriever), "Explanation message");
            }

            var thing = new IExistsForTestingValueRetrieving();
            service.GetValueRetrievers().Register(thing);
            Assert.Equal(1, service.GetValueRetrievers().Size(), "Explanation message");
            Assert.Same(thing, service.GetValueRetrievers().Get(0), "Explanation message");
        }
    }
}
